catalog/transformer/catalog-transformer-csv-common/src/test/java/ddf/catalog/transformer/csv/common/CsvTransformerTest.java

```diff
@@ -188,17 +188,14 @@ public void filterEmptyValueAttributes() {
       metacard.setAttribute(a);
     }
     metacard.setAttribute(new AttributeImpl("attribute9", ""));
-    List<Metacard> metacards = metacardList.stream().collect(Collectors.toList());
+    List<Metacard> metacards = new ArrayList<>(metacardList);
     metacards.add(metacard);
     Set<AttributeDescriptor> nonEmptyValues = CsvTransformer.getNonEmptyValueAttributes(metacards);
 
     assertThat(nonEmptyValues, hasSize(6));
-    final Optional<AttributeDescriptor> attributeDescriptorOptional =
-        nonEmptyValues.stream().findFirst();
-    assertThat(attributeDescriptorOptional.isPresent(), is(true));
-
-    final AttributeDescriptor attributeDescriptor = attributeDescriptorOptional.get();
-    assertThat(attributeDescriptor, notNullValue());
+    for (AttributeDescriptor descriptor: nonEmptyValues){
+      assertThat(metacardList.stream().anyMatch(m -> m.getAttribute(descriptor.getName()) != null), is(true));
```

Comment:
We also need to verify the attribute values are not nulls or empty strings.

However, I think there is a better way to do this. We should be explicit about which attributes should not be in this collection and assert on them, since we know exactly which ones they are. If I am understanding this test correctly, `atttribute7`, `attribute8`, and `attribute9` should not be included.
Something like this:
```java
Set<String> nonEmptyAttributes = nonEmptyValues.stream().map(AttributeDescriptor::getName).collect(toSet());
assertThat(nonEmptyAttributes, not(hasItem("attribute7")));
...
```

Reply:
fixed as suggested.